diff --git a/build/tasks/kernel.mk b/build/tasks/kernel.mk
index 3f679028..fd88e4bf 100644
--- a/build/tasks/kernel.mk
+++ b/build/tasks/kernel.mk
@@ -673,7 +673,27 @@ ifeq ($(BOARD_USES_QCOM_MERGE_DTBS_SCRIPT),true)
 	$(hide) find $(DTBS_OUT) -type f -name "*.dtb*" | xargs rm -f
 	mv $(DTB_OUT)/arch/$(KERNEL_ARCH)/boot/dts/vendor/*/*.dtb $(DTB_OUT)/arch/$(KERNEL_ARCH)/boot/dts/vendor/*/*.dtbo $(DTBS_BASE)/
 	PATH=$(abspath $(HOST_OUT_EXECUTABLES)):$${PATH} python3 $(BUILD_TOP)/vendor/lineage/build/tools/merge_dtbs.py --base $(DTBS_BASE) --techpack $(DTB_OUT)/arch/$(KERNEL_ARCH)/boot/dts/vendor/qcom --out $(DTBS_OUT)
-	cat $(shell find $(DTBS_OUT) -type f -name "${TARGET_MERGE_DTBS_WILDCARD}.dtb" | sort) > $@
+	#cat $(shell find $(DTBS_OUT) -type f -name "${TARGET_MERGE_DTBS_WILDCARD}.dtb" | sort) > $@
+	cat $(DTBS_OUT)/cliffs7.dtb > $@
+	cat $(DTBS_OUT)/cliffs7p.dtb >> $@
+	cat $(DTBS_OUT)/cliffs.dtb >> $@
+	cat $(DTBS_OUT)/cliffsp.dtb >> $@
+	cat $(DTBS_OUT)/pineapple.dtb >> $@
+	cat $(DTBS_OUT)/pineapplep.dtb >> $@
+	cat $(DTBS_OUT)/pineapplep-sg-v2.dtb >> $@
+	cat $(DTBS_OUT)/pineapplep-v2.dtb >> $@
+	cat $(DTBS_OUT)/pineapple-v2.dtb >> $@
+	cat $(DTBS_OUT)/volcano6.dtb >> $@
+	cat $(DTBS_OUT)/volcano6i-fp1.dtb >> $@
+	cat $(DTBS_OUT)/volcano6i-fp2.dtb >> $@
+	cat $(DTBS_OUT)/volcano6i-fp3.dtb >> $@
+	cat $(DTBS_OUT)/volcano6i-fp4.dtb >> $@
+	cat $(DTBS_OUT)/volcano6ip-fp1.dtb >> $@
+	cat $(DTBS_OUT)/volcano6ip-fp2.dtb >> $@
+	cat $(DTBS_OUT)/volcano6ip-fp3.dtb >> $@
+	cat $(DTBS_OUT)/volcano6ip-fp4.dtb >> $@
+	cat $(DTBS_OUT)/volcano6p.dtb >> $@
+	cat $(DTBS_OUT)/volcano.dtb >> $@
 else
 	cat $(shell find $(DTB_OUT)/arch/$(KERNEL_ARCH)/boot/dts -type f -name "*.dtb" | sort) > $@
 endif # BOARD_USES_QCOM_MERGE_DTBS_SCRIPT
diff --git a/config/BoardConfigKernel.mk b/config/BoardConfigKernel.mk
index 3a177b44..45366118 100644
--- a/config/BoardConfigKernel.mk
+++ b/config/BoardConfigKernel.mk
@@ -215,7 +215,7 @@ ifeq (true,$(filter true, $(TARGET_NEEDS_DTBOIMAGE) $(BOARD_KERNEL_SEPARATED_DTB
 endif
 
 # Set the default dtb target
-TARGET_KERNEL_DTB ?= dtbs
+TARGET_KERNEL_DTB ?= dtbs DTC_FLAGS=-@
 
 # Set no external modules by default
 TARGET_KERNEL_EXT_MODULE_ROOT ?=
